packages/shared/components/StakingAssetTile.svelte

```diff
@@ -0,0 +1,246 @@
+    let tooltipText: TooltipText
+    let remainingTime: number
+
+    $: assetIconColor = isBright(asset?.color) ? 'gray-800' : 'white'
+    $: isDarkModeEnabled = $appSettings.darkMode
+    $: isActivelyStaking = _getAccount($stakedAccounts) && isStakingPossible($stakingEventState)
+    $: isPartiallyStaked = _getAccount($partiallyStakedAccounts) && isStakingPossible($stakingEventState)
+    $: hasStakingEnded = $stakingEventState === ParticipationEventState.Ended
+    $: $participationOverview, (tooltipText = getLocalizedTooltipText())
+    $: remainingTime = asset?.name === Token.Assembly ? $assemblyStakingRemainingTime : $shimmerStakingRemainingTime
+    $: {
+        if (hasAccountReachedMinimumAirdrop($selectedAccount) && !isStakingPossible($stakingEventState)) {
+            isBelowMinimumRewards = false
+        } else {
+            const accountOverview = $participationOverview.find(
+                (accountParticipationOverview) => accountParticipationOverview.accountIndex === $selectedAccount?.index
+            )
+            isBelowMinimumRewards = accountOverview?.[`${asset?.name}RewardsBelowMinimum`] > 0
+        }
+    }
+    $: canShowWarningState =
+        isPartiallyStaked ||
+        (isBelowMinimumRewards && !_getAccount($stakedAccounts) && isStakingPossible($stakingEventState)) ||
+        (isBelowMinimumRewards && hasStakingEnded)
+
+    function toggleTooltip(): void {
+        canShowTooltip = !canShowTooltip
+    }
+    function handleTileClick(): void {
+        openPopup({ type: 'airdropNetworkInfo', props: { airdrop } })
+    }
+    function _getAccount(accounts: WalletAccount[]): WalletAccount {
+        return accounts.find((account) => account.alias === $selectedAccount.alias)
+    }
+    function getLocalizedTooltipText(): TooltipText {
+        if (isPartiallyStaked) {
+            return {
+                title: localize('tooltips.partiallyStakedFunds.title', {
+                    values: { amount: formatUnitBestMatch(getUnstakedFunds($selectedAccount)) },
+                }),
+                body: [localize('tooltips.partiallyStakedFunds.body')],
+            }
+        } else if (isBelowMinimumRewards) {
+            const minimumRewards = getFormattedMinimumRewards(airdrop)
+            if (hasStakingEnded) {
+                return {
+                    title: localize('tooltips.stakingMinRewards.title'),
+                    body: [
+                        localize('tooltips.stakingMinRewards.bodyDidNotReachMin', {
+                            values: {
+                                airdrop: capitalize(airdrop),
+                                airdropRewardMin: minimumRewards,
+                            },
+                        }),
+                    ],
+                }
+            } else if (!isAccountStaked($selectedAccount?.id) && isStakingPossible($stakingEventState)) {
+                const timeNeeded = <number>getTimeUntilMinimumAirdropReward($selectedAccount, airdrop)
```

Comment:
This smells like getTimeUntilMinimumAirdropReward couldn't make a decision to return a number only. Why is this? Maybe it needs changing. I think functions should know whether to return a string or a number, it's their responsibility, not yours to convert the type